### DISCUSSION TITLE
[Shipment Management] [Zwroty] Pobieranie kodu kreskowego na etykiecie oraz operowanie na zwrotach

### DISCUSSION THREAD

**Discussion by macierak** [allegro/allegro-api#10545] [Suggestions - resolved]
Cześć!

Do mojej aplikacji potrzebuję pobierać wartość kodów kreskowych prosto z API - Głównie problem otrzymaliśmy dla zamówień przez DPD.

```
"packages": [
{
"waybill": "string",
"barcode": "string", //nowe pole
"type": "DOX",
(...)
```
Czy była by opcja dodać taką wartość?

Druga rzecz jaką napotkałem przy operacji na zwrotach:
Na ten moment, przy pobieraniu zwrotów dostajemy jedynie wartość orderId do namierzenia danej przesyłki.
W momencie kiedy z powodu błędu systemu zostaną utworzone dodatkowe przesyłki, może zrobić się zamieszanie.
Dodanie pola shipmentId oryginalnej przesyłki mogłoby umożliwić znalezienie konkretnego zamówienia kuriera.

```
"customerReturns": [
{
"id": "a3405c27-b01c-4357-9bea-e13925708b46",
"createdAt": "2020-01-11T09:36:57.00Z",
"referenceNumber": "1234/Z04A",
"orderId": "a3405c27-b01c-4357-9bea-e13925708b46",
"originalShipmentId":"UUID"
(...)
```

**Comment by AureliuszBrussy** [comment]
@macierak przekonwertowałem Twoje zgłoszenie do [Discussions](https://github.com/allegro/allegro-api/discussions), gdzie zbieramy wszystkie sugestie zmian w API. Równocześnie dziękuję za Twoją sugestię - Jeśli zdecydujemy się wprowadzić takie rozwiązania, poinformujemy w wątku.

**Reply by macierak** [reply]
Super!
Dzięki wielkie :D

**Comment by axandar** [comment]
@AureliuszBrussy dorzucę jeszcze od siebie informację. Od strony magazynów co mają zautomatyzowane skanowanie paczek, bardzo ciężko jest rozpoznać kody kreskowe z etykiet. Mają one często dodane znaki na początku lub końcu i przy obsłudze wielu firm kurierskich zaczynają być problemy z rozpoznawaniem. Jestem pewien że informacje o tym jakie jest list przewozowy i kod kreskowy na tym liście będzie przydatna dla wielu firm podczas prób integracji danych z Allegro do wewnętrznych systemów logistycznych.

**Reply by AureliuszBrussy** [reply]
Dzięki za dodatkowy feedback, zanotowane.

**Comment by MaciejFrackowiak** [comment]
W zasobie do obsługi przesyłek [dodaliśmy informację](https://developer.allegro.pl/news/wysylam-z-allegro-wprowadzilismy-zmiany-na-zasobach-do-zarzadzania-wysylka-przesylek-i-ich-odbiorem-przez-kuriera-oADdP41WVHA) o firmie i numerze trackingowym operatora pośredniczącego w realizacji przesyłki. Natomiast nie przewidujemy dodawania do zwrotu informacji o przesyłkce pierwotnej - tu mamy osobny [endpoint](https://developer.allegro.pl/documentation/#tag/Order-management/operation/getOrderShipmentsUsingGET), który zwraca wszystkie numery trackingowe dodane do zamówienia.
